source/beamline/CLS/CLSShutters.cpp

```diff
@@ -14,7 +19,88 @@ CLSShutters::CLSShutters(const QString &name, QObject *parent) :
 
 CLSShutters::~CLSShutters()
 {
+	shuttersBeamOnOrderMap_.clear();
+}
+
+AMListAction3* CLSShutters::createBeamOnActionList()
```

Comment:
I think your solution should probably replace the existing `moveToOpen/ClosedAction()` functionality.